Add global carbon budget

pabloarosado opened the pull request:
Add Meadow, Garden and Grapher steps for Global Carbon Budget data.
The resulting dataset contains all variables from [the old importers `gcp_gcb` dataset](https://github.com/owid/importers/tree/master/gcp_gcb), plus the additional variables from [this ETL step](https://github.com/owid/etl/tree/master/etl/steps/data/garden/gcp/2022-09-29), plus some new variables given in the data (national land-use change emissions).

@edomt if you have a few minutes, could you please check the garden metadata file? (etl/steps/data/garden/gcp/2022-11-11/global_carbon_budget.meta.yml) There's no need to go line by line, which would take a long time, but simply check some entries at random to search for any possible inconsistencies (since filling up the metadata was a semi-manual process). Thanks.

Before merging this PR, the following steps need to be completed:
* [x] Merge https://github.com/owid/walden/pull/127 (until that one is merged, CI in this PR will fail)

edomt reviewed (changes requested):
Thanks @pabloarosado! I've checked 10 entries at random:

- `global_cumulative_emissions_from_gas` has an empty `description`, as well as several other entries above and below it – I'm not sure if this is expected.
- Same for `consumption_emissions_as_share_of_global`
- The `title` of the global cumulative variables takes the form of `Global annual cumulative CO2 emissions from <x>`. Given that these are cumulative, we should drop the `annual` for clarity.
- `land_use_change_quality_flag` has the wrong units (tonnes and t). It should probably have empty units.

pabloarosado commented:
> Thanks @pabloarosado! I've checked 10 entries at random:
> 
> * `global_cumulative_emissions_from_gas` has an empty `description`, as well as several other entries above and below it – I'm not sure if this is expected.
> * Same for `consumption_emissions_as_share_of_global`
> * The `title` of the global cumulative variables takes the form of `Global annual cumulative CO2 emissions from <x>`. Given that these are cumulative, we should drop the `annual` for clarity.
> * `land_use_change_quality_flag` has the wrong units (tonnes and t). It should probably have empty units.

Thanks a lot @edomt . "Global" variables are not used anywhere. I just added them because, in the garden dataset, it may be useful to have them as reference on how “share” variables were calculated. But they are actually a bit redundant. So I’ve left them without a description. Regarding the other suggestions, in the last commit I have:
* Fixed wrong units (of GDP and land-use quality flag).
* Changed “global annual cumulative” to “global cumulative”.
* Changed CO2 to CO₂.
* Removed “Global coal emissions include all country emissions” (and idem for other sources). We had this sentence in the variable descriptions of the previous release, but I don’t think it makes much sense. Global emissions from coal do not include all country emissions (they include only emissions from coal). Maybe it was meant to say “all countries emissions”. I found the sentence confusing, so I’ve removed it from all descriptions.

bnjmacdonald reviewed (approved):
Looks good! Thank you for all of the very well-commented code, made it much easier to review. 

None of my comments/suggestions are necessary to address before merging.

pabloarosado commented on etl/steps/data/meadow/gcp/2022-11-11/global_carbon_budget_fossil_co2_emissions.py:
```diff
@@ -0,0 +1,49 @@
+    #
+    # Set an appropriate index and sort conveniently.
+    emissions_df = emissions_df.set_index(["Country", "Year"], verify_integrity=True).sort_index().sort_index(axis=1)
+
```
In principle, the Meadow step should involve the minimum processing possible, just to adapt the source data into a convenient dataframe format with metadata (aka a table). But sometims it's unclear what should or should not be done at this stage.
Underscoring is done a bit later in line 45, by `underscore_table`.

pabloarosado commented on etl/steps/data/meadow/gcp/2022-11-11/global_carbon_budget_fossil_co2_emissions.py:
```diff
@@ -0,0 +1,49 @@
+    # Process data.
+    #
+    # Set an appropriate index and sort conveniently.
+    emissions_df = emissions_df.set_index(["Country", "Year"], verify_integrity=True).sort_index().sort_index(axis=1)
```
I think we don't need to check so much for data quality at the Meadow step, and leave all those checks for Garden. I don't think the `verify_integrity` is even necessary. But we can talk about this at some point, maybe it's something we should consider.

pabloarosado commented on etl/steps/data/grapher/gcp/2022-11-11/global_carbon_budget.py:
```diff
@@ -0,0 +1,56 @@
+    # Create additional variables in the table that have nans filled with zeros (for two specific stacked area charts).
+    for variable in VARIABLES_TO_FILL_WITH_ZEROS:
+        new_variable_name = variable + "_zero_filled"
+        table[new_variable_name] = table[variable].fillna(0)
```
Good idea. I just added a sanity check to ensure that all countries span all years (and they did).

pabloarosado commented:
Thanks @bnjmacdonald for review and suggestions!
